src/expire.c

```diff
@@ -294,10 +346,18 @@ void activeExpireCycle(int type) {
             int origin_ttl_samples = data.ttl_samples;
 
             while (data.sampled < num && checked_buckets < max_buckets) {
-                db->expires_cursor = kvstoreScan(db->expires, db->expires_cursor, -1, expireScanCallback,
-                                                 isExpiryTableValidForSamplingCb, &data);
-                if (db->expires_cursor == 0) {
-                    db_done = 1;
+                hashtableScanFunction scan_cb = jobType == KEYS ? expireScanCallback : fieldExpireScanCallback;
```

Comment:
nit: I find ternaries much more readable if the expression part is parenthesized.
```suggestion
                hashtableScanFunction scan_cb = (jobType == KEYS) ? expireScanCallback : fieldExpireScanCallback;
```

This doesn't need to be in the loop, but I suspect the compiler will optimize it out of the loop anyway.  Probably better readability with it here.